Fix `dev.num_executions` bug with QNode caching (#2171)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>Bug fixes</h3>

> * Fixes a bug where an incorrect number of executions are recorded by
>   a QNode using a custom cache e.g., for `diff_method="backprop"`.
>   [(#2171)](https://github.com/PennyLaneAI/pennylane/pull/2171)

Reviewer: ```suggestion
* Fixes a bug where an incorrect number of executions are recorded by
  a QNode using a custom cache with `diff_method="backprop"`.
  [(#2171)](https://github.com/PennyLaneAI/pennylane/pull/2171)
```
